### ISSUE TITLE
HikariCP 2.3.10

### ISSUE THREAD

**Issue by dannygu** [brettwooldridge/HikariCP#436]
Hi,

I recently upgraded to 2.3.10 since then i keep seeing the following in my logs:

2015-09-30 06:41:54,164 [http-apr-11910-exec-33] DEBUG com.zaxxer.hikari.proxy.ConnectionProxy - HikariPool-0 Resetting dirty on com.mysql.jdbc.JDBC4Connection@7487fd38 (readOnlyDirty=false,autoCommitDirty=false,isolationDirty=false,catalogDirty=true)
2015-09-30 06:41:54,283 [http-apr-11910-exec-9] DEBUG com.zaxxer.hikari.proxy.ConnectionProxy - HikariPool-0 Resetting dirty on com.mysql.jdbc.JDBC4Connection@5b019393 (readOnlyDirty=false,autoCommitDirty=false,isolationDirty=false,catalogDirty=true)
2015-09-30 06:41:54,284 [http-apr-11910-exec-9] DEBUG com.zaxxer.hikari.proxy.ConnectionProxy - HikariPool-0 Resetting dirty on com.mysql.jdbc.JDBC4Connection@5b019393 (readOnlyDirty=false,autoCommitDirty=false,isolationDirty=false,catalogDirty=true)
2015-09-30 06:41:54,284 [http-apr-11910-exec-9] DEBUG com.zaxxer.hikari.proxy.ConnectionProxy - HikariPool-0 Resetting dirty on com.mysql.jdbc.JDBC4Connection@5b019393 (readOnlyDirty=false,autoCommitDirty=false,isolationDirty=false,catalogDirty=true)
2015-09-30 06:41:54,547 [http-apr-11910-exec-30] DEBUG com.zaxxer.hikari.proxy.ConnectionProxy - HikariPool-0 Resetting dirty on com.mysql.jdbc.JDBC4Connection@b380b2 (readOnlyDirty=false,autoCommitDirty=false,isolationDirty=false,catalogDirty=true)

Can you please explain what is this about ? not sure this is actually an error ? 

Thanks!

**Comment by dannygu** [comment]
FYI - 2.4.1 gives me the folllwing:

2015-09-30 06:53:35,139 [http-apr-11920-exec-31] DEBUG com.zaxxer.hikari.pool.PoolElf - HikariPool-0 - Reset (nothing) on connection com.mysql.jdbc.JDBC4Connection@2c9335cb
2015-09-30 06:53:35,140 [http-apr-11920-exec-31] DEBUG com.zaxxer.hikari.pool.PoolElf - HikariPool-0 - Reset (nothing) on connection com.mysql.jdbc.JDBC4Connection@2c9335cb
2015-09-30 06:53:35,140 [http-apr-11920-exec-31] DEBUG com.zaxxer.hikari.pool.PoolElf - HikariPool-0 - Reset (nothing) on connection com.mysql.jdbc.JDBC4Connection@2c9335cb
2015-09-30 06:53:35,150 [http-apr-11920-exec-31] DEBUG com.zaxxer.hikari.pool.PoolElf - HikariPool-0 - Reset (catalog) on connection com.mysql.jdbc.JDBC4Connection@2c9335cb
2015-09-30 06:53:35,911 [http-apr-11920-exec-4] DEBUG com.zaxxer.hikari.pool.PoolElf - HikariPool-0 - Reset (nothing) on connection com.mysql.jdbc.JDBC4Connection@d38e31b

is this the same as above ?

**Comment by brettwooldridge** [comment]
@dannygu Yes, the second is the same as the first.

This would be triggered by a framework (Hibernate, etc.) or your application itself, for example, calling ``connection.setAutoCommit(false)`` when the pool is already configured as autoCommit=false.

Why is this logged?  Well, many drivers (for example MySQL) will by default perform a roundtrip to the database whenever ``setAutoCommit()`` is called.  Even if the driver could know that it is a *no-op* it will do so.  For example:
```java
connection.setAutoCommit(true);
connection.setAutoCommit(true);
```
will result in two roundtrips to the database, even though the second is obviously unnecessary.  In the case of MySQL, the driver can be configured to cache connection state and ignore the second call, but that is not the default behavior.

In a nutshell, this log is saying something called ``setReadOnly()``, ``setAutoCommit()``, ``setTransactionIsolation()``, and ``setCatalog()``, but the values were actually the same as the pool connection defaults (*"Reset (nothing) on connection ..."*) -- so this resulted in 4 wasted roundtrips to the database.
